src/main/java/hudson/scm/subversion/WorkspaceUpdater.java

```diff
@@ -190,15 +190,6 @@ protected SVNRevision getRevision(ModuleLocation l) {
             return r;
         }
 
-        /**
-         * Returns {@link org.tmatesoft.svn.core.SVNDepth} by string value.
-         *
-         * @return {@link org.tmatesoft.svn.core.SVNDepth} value.
-         */
-        protected static SVNDepth getSvnDepth(String name) {
```

Comment:
IIUC it breaks the binary compatibility as it confirmed by the changes above.
WorkspaceUpdater implementations may exist outside the plugin, so it needs to be handled carefully

Reply:
I don't fully understand what contract we need to provide here. Is it:
1) we just need to continue providing any accessible methods, that people might be using so that we don't break stuff.
2) we need to do 1 + continue to call the functions in a similar way to before so that they can be used as an extension point. 

If 1 then obviously I just need to add the method back. Should I also mark it as deprecated. 
If 2 then I'll also have to change all my getSvnDepthFor* function to getDepthOptionFor* and wrap their invocations in calls the the getSvnDepth function. 

Which do you think is better?

Reply:
Jenkins project uses the approach number 1, because we cannot guarantee that plugins maintainers deliver the required fixes for N + 1 and then that the users do updates properly. Breaking changes are being delivered only in major releases (first-digit change), but there is no justification of doing such release in the SVN plugin while it has no active maintainer